parser/parser.go

```diff
@@ -3,16 +3,24 @@ package parser
 import (
 	"bufio"
 	"bytes"
+	"crypto/sha256"
 	"errors"
 	"fmt"
 	"io"
+	"net/http"
+	"os"
+	"path/filepath"
 	"strconv"
 	"strings"
 
 	"golang.org/x/text/encoding/unicode"
 	"golang.org/x/text/transform"
+
+	"github.com/ollama/ollama/api"
 )
 
+var ErrModelNotFound = errors.New("no Modelfile or safetensors files found")
+
 type File struct {
```

Comment:
```suggestion
type Modelfile struct {
```
might be a better name for this